### DISCUSSION TITLE
Mac Touch Bar Support

### DISCUSSION THREAD

**Discussion by MsBlockCat** [opentoonz/opentoonz#6705] [Ideas]
On MacBook Pros since 2016, we've had a Touch Bar, which is basically a nice touch screen that replaces the FN keys. That touch screen can do different things depending on what app you're in and what you're doing in that app.

It would be great if OpenToonz supported the Touch Bar! Even Adobe Animate doesn't support it yet. I have some different ideas for how it could work, but we'd probably want to be able to customize it as you can in some other apps, by choosing View > Customize Touch Bar... And then using macOS's built-in interface to choose which buttons and widgets are on it. Some buttons/widgets I'm hoping we could have as options are a Play/Pause Animation button, Selection Tool button, Control Point Editor button, Fast Render to MP4... button, and a widget slider to go to different parts of the animation quickly, like is built in when you play a video in Safari on Macs with a Touch Bar (see below screenshot).
<img width="1085" alt="Touch Bar Shot 2020-03-02 at 10 15 51 PM" src="https://user-images.githubusercontent.com/37186692/75739626-645ea300-5cd3-11ea-8542-715431e27b74.png">

I'm not sure how hard this would be to add, but even just adding a Play/Pause Animation button would make a big difference!

**Comment by artisteacher** [comment]
I would love to see that too! The simplest thing to start with would probably be to get the Touch Bar to use the command bar - it's already customizable...

Unfortunately, there is not currently a mac focused developer. Perhaps if there was a large enough bounty, someone would be interested.

**Comment by MsBlockCat** [comment]
Good thinking! The only problem I see with the command bar is that its icons are pixelated, but that's a separate issue and relatively minor.

**Comment by piotrnikov** [comment]
Hello!
For those working on MacOs you can use an app called BetterTouch tools (https://folivora.ai/). It's awesome and basically gives you the ability to configure the TouchBar (amongst other things) however you like, whichever software you use.
Try it you won't regret it.
Piotr

**Comment by MsBlockCat** [comment]
Another good use of the Touch Bar I just thought might be to, when you're in the Preferences, change the Touch Bar from showing the Command Bar to showing all the Preferences tabs so you can more easily switch between them!

**Comment by MsBlockCat** [comment]
@RodneyBaker I've made a $50 bounty for this! https://www.bountysource.com/issues/89477264-mac-touch-bar-support
